wiring/inc/spark_wiring_cellular.h

```diff
@@ -75,6 +79,9 @@ class CellularClass : public NetworkClass
     void setListenTimeout(uint16_t timeout) {
         network_set_listen_timeout(*this, timeout, NULL);
     }
+    inline void setListenTimeout(particle::chrono::seconds s) { setListenTimeout(std::min(s.count(), static_cast<system_tick_t>(std::numeric_limits<uint16_t>::max()))); }
```

Comment:
I would avoid manually specifying overloads for all the duration types as that increases the manual work, as well as (as you might have seen in the Travis logs) introduces overload ambiguity due to automatic casts. The common practice for `std::chrono` durations is either use a template or use a minimum resolution that a function may take:

```cpp
template<typename Rep, typename Period>
void somefunc(std::chrono::duration<Rep, Period> arg);
```

This should still work for seconds, minutes, hours etc.:
```cpp
void delay(particle::chrono::milliseconds arg);
```

Reply:
I didn't realize it would do the casting for me, I'm removing the unnecessary API. I will also prototype templating (according to our side conversation).

Reply:
The minimum resolution implementation has been pushed for review. I will be prototyping templatization.

If the template is able to automatically resolve without needing to be specified, then I will replace the current implementation. However, I have low confidence this will occur, because I had originally attempted to implement the API in this way.

Either way, I will reply in this thread.

Reply:
I forgot to confirm the templated method works nicely (good syntax) as @avtolstoy had expected.

However it has the undesired effect of allowing for overflow conditions that we must either consider or ignore. By using the minimum resolution implementation we are able to bubble that decision up to the user requesting the duration.

I ran into difficulty when I was required to template the constructor of the `ApplicationWatchdog` class, so I fell back to the minimum resolution implementation.